curriculum/challenges/english/blocks/lecture-user-centered-design/672bb02ecb230779bbbaccd9.md

```diff
@@ -5,24 +5,457 @@ challengeType: 19
 
+:::interactive_editor
+
+```html
+<link rel="stylesheet" href="styles.css" />
+
+<div class="shopping-cart">
+  <button id="cart-btn" aria-label="Shopping cart">
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      width="24"
+      height="24"
+      fill="none"
+      stroke="currentColor"
+      stroke-width="2"
+      stroke-linecap="round"
+      stroke-linejoin="round"
+      class="feather feather-shopping-cart"
+      viewBox="0 0 24 24"
+      aria-hidden="true"
+      focusable="false"
+    >
+      <circle cx="9" cy="21" r="1"></circle>
+      <circle cx="20" cy="21" r="1"></circle>
+      <path d="M1 1h4l2.68 13.39a2 2 0 0 0 2 1.61h9.72a2 2 0 0 0 2-1.61L23 6H6"></path>
+    </svg>
+    <span id="item-count">2</span>
+  </button>
+</div>
+```
+
+```css
+.shopping-cart {
+  position: fixed;
+  top: 20px;
+  right: 20px;
+  font-family: sans-serif;
+}
+
+#cart-btn {
+  background: none;
+  border: none;
+  cursor: default;
+  position: relative;
+  display: flex;
+  align-items: center;
+  padding: 4px;
+  color: #333;
+  font-size: 24px;
+}
+
+#item-count {
+  background: red;
+  color: white;
+  border-radius: 50%;
+  padding: 2px 6px;
+  font-size: 13px;
+  position: absolute;
+  top: -6px;
+  right: -6px;
+  min-width: 18px;
+  height: 18px;
+  line-height: 14px;
+  text-align: center;
+  font-weight: 700;
+  user-select: none;
+  box-sizing: border-box;
+}
+
+#cart-details {
+  background: white;
+  border: 1px solid #ccc;
+  max-height: 300px;
+  overflow-y: auto;
+  box-shadow: 0 4px 8px rgba(0,0,0,0.1);
+  margin-top: 10px;
+  padding: 10px;
+  position: relative;
+  z-index: 1000;
+}
+
+#cart-items {
+  list-style: none;
+  padding: 0;
+  margin: 0;
+}
+
+#cart-items li {
+  padding: 5px 0;
+  border-bottom: 1px solid #eee;
+}
+```
+
+:::
+
 Another consideration is providing a clear way for users to update the quantity of items in their cart. This can be done by providing a quantity input field next to each item in the cart. Users can easily update the quantity of an item by changing the number in the input field.
 
 You should also provide a "Remove" button next to each item in the cart. This allows users to easily remove items from their cart. You don't want to make it difficult for users to remove items from their cart, as this can lead to frustration and abandoned carts.
 
+:::interactive_editor
+
+```html
+<link rel="stylesheet" href="styles.css" />
+
+<div class="shopping-cart">
+  <h2>Your Cart</h2>
+  <ul class="cart-items">
+    <li class="cart-item">
+      <span class="item-name">T-Shirt</span>
+      <input type="number" min="1" value="2" class="quantity-input" aria-label="Quantity for T-Shirt" />
+      <button class="remove-btn" aria-label="Remove T-Shirt from cart">Remove</button>
```

Comment:
The `aria-label` is redundant because the button already has a visible label ("Remove").

```suggestion
      <button class="remove-btn">Remove</button>
```